pkg/chunk/local/boltdb_index_client_test.go

```diff
@@ -0,0 +1,117 @@
+package local
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"go.etcd.io/bbolt"
+)
+
+var (
+	testKey   = []byte("test-key")
+	testValue = []byte("test-value")
+)
+
+func setupDb(t *testing.T, boltdbIndexClient *BoltIndexClient, dbname string) {
+	db, err := boltdbIndexClient.GetDB(dbname, DBOperationWrite)
+	require.NoError(t, err)
+
+	err = db.Update(func(tx *bbolt.Tx) error {
+		b, err := tx.CreateBucketIfNotExists(bucketName)
+		if err != nil {
+			return err
+		}
+
+		return b.Put(testKey, testValue)
+	})
+	require.NoError(t, err)
+}
+
+func TestBoltDBReload(t *testing.T) {
+	dirname, err := ioutil.TempDir(os.TempDir(), "boltdb")
+	require.NoError(t, err)
+
+	indexClient, err := NewBoltDBIndexClient(BoltDBConfig{
+		Directory: dirname,
+	})
+	require.NoError(t, err)
+
+	testDb1 := "test1"
+	testDb2 := "test2"
+
+	boltdbIndexClient := indexClient
+	setupDb(t, boltdbIndexClient, testDb1)
+	setupDb(t, boltdbIndexClient, testDb2)
+
+	boltdbIndexClient.reload()
+	require.Equal(t, 2, len(boltdbIndexClient.dbs), "There should be 2 boltdbs open")
+
+	require.NoError(t, os.Remove(filepath.Join(dirname, testDb1)))
+
+	droppedDb, err := boltdbIndexClient.GetDB(testDb1, DBOperationRead)
+	require.NoError(t, err)
+
+	valueFromDb := []byte{}
+	_ = droppedDb.View(func(tx *bbolt.Tx) error {
+		b := tx.Bucket(bucketName)
+		valueFromDb = b.Get(testKey)
+		return nil
+	})
+	require.Equal(t, testValue, valueFromDb, "should match value from db")
+
+	boltdbIndexClient.reload()
+
+	require.Equal(t, 1, len(boltdbIndexClient.dbs), "There should be 1 boltdb open")
```

Comment:
May we also check if `boltdbIndexClient.GetDB(testDb1, DBOperationRead)` returns `ErrUnexistentBoltDB`?

_Same comment applies to the calls to `<bolt db>.Stop()` in the tests you've added_

Reply:
I am sorry if I didn't understand it, is this what you are looking for?
https://github.com/cortexproject/cortex/pull/2049/files/273032a393eb1a6cc815e5387a1bd2259f494dc4#diff-4f420894f40d4e2e1a9b491716cc37dfR96

Reply:
I mean trying to `GetDB(testDb1)` and then `require.Equal(t, ErrUnexistentBoltDB, err)`